lua/rocks-config/init.lua

```diff
@@ -99,8 +99,39 @@ local function check_for_duplicate(plugin_name, config_basename, mod_name)
     end
 end
 
+---Load a config and register any errors that happened while trying to load it.
+---Returns false if the module was not found and true if it was, even if errors happened.
+---@param plugin_name string The plugin that is being configured
+---@param config_basename string The basename of the configuration module.
+---@param mod_name string The configuration module to load.
+---@return boolean
+local function load_config(plugin_name, config_basename, mod_name)
+    local ok, result = pcall(function()
+        return try_load_config(mod_name)
+    end)
+
+    if not ok then
+        -- Module was found but failed to load.
+        table.insert(rocks_config.failed_to_load, { plugin_name, config_basename, result })
+        return true
+    end
+
+    if type(result) ~= "boolean" then
+        error("try_load_config did not return boolean as expected.")
```

Comment:
suggestion: If a user sees this error, it will be very confusing. How about something like:

```suggestion
        error("rocks-config.nvim: The impossible happened! Please report this bug: try_load_config did not return boolean as expected.")
```